tests/ftests/051-sudo-set_uid_gid_v2.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# Test to set the uid/gid on a cgroup using the python bindings
+#
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Author: Tom Hromatka <tom.hromatka@oracle.com>
+#
+
+from cgroup import Cgroup as CgroupCli, CgroupVersion
+from libcgroup import Cgroup, Version
+import consts
+import ftests
+import utils
+import sys
+import os
+
+CGNAME = '051setuidgid'
+CONTROLLER = 'cpu'
+TASKS_UID = 1234
+TASKS_GID = 5678
+CTRL_UID = 3456
+CTRL_GID = 7890
+
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if CgroupVersion.get_version(CONTROLLER) != CgroupVersion.CGROUP_V2:
+        result = consts.TEST_SKIPPED
+        cause = 'This test requires cgroup v2'
+
+    return result, cause
+
+
+def setup(config):
+    return consts.TEST_PASSED, None
+
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    cg = Cgroup(CGNAME, Version.CGROUP_V2)
+    cg.set_uid_gid(TASKS_UID, TASKS_GID, CTRL_UID, CTRL_GID)
+    cg.add_controller(CONTROLLER)
+    cg.create(ignore_ownership=False)
+
+    ctrl_path = os.path.join(CgroupCli.get_controller_mount_point(CONTROLLER), CGNAME,
+                             'cgroup.procs')
+
+    uid = utils.get_file_owner_uid(config, ctrl_path)
+    if uid != CTRL_UID:
+        result = consts.TEST_FAILED
+        cause = 'Expected cgroup.procs owner to be {} but it\'s {}'.format(CTRL_UID, uid)
+
```

Comment:
Is the fallthrough intentional? If both `uid` and `gid` checks fails,  then the `gid`  will override the `uid cause`

Reply:
Honestly I didn't really think it through too hard.  I made sure that success couldn't accidentally overwrite a failure, but I didn't worry too much about a failure overwriting another failure.  I'll see if I can concatenate the failures into a single `cause`